### PR TITLE
Linting pre-commit hook

### PR DESCRIPTION
**Pull request by ismirlia**
Added pre-commit hook to ensure linting is checked before the user commits.

### What are the issues this pull addresses (issue numbers / links)?
N/A

### Did you add tests to cover your changes (yes/no)?
no
### Did you update the documentation accordingly (yes/no)?
no
### Did you read the CONTRIBUTING document (yes/no)?
no
### Details and comments
We may need to update the installation instruction to make users aware and allow opt-in.

### CODE REVIEW

**Comment by ThomasGM4** [comment]
Did this get some test runs on different systems like you were looking for Axel?

**Review by ThomasGM4 — DISMISSED** [review]
Isn't working in Github Desktop for Windows atm (Axel will investigate later), but works from command lines just fine.

File: hook_setup.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+ln -s -f ../../hooks/pre-commit .git/hooks/pre-commit
```
[the remark this answers is not in the document]
Comment (priti-ashvin-shah-ibm) [reply hook_setup.sh:3]: Agreed to request users to use Git Bash shell on windows.

[the remark this answers is not in the document]
Comment (priti-ashvin-shah-ibm) [reply hook_setup.sh:3]: LGTM

**Review by priti-ashvin-shah-ibm — APPROVED** [review]
LGTM

**Pull request merged**